sdk/src/main/java/com/horizen/account/api/rpc/service/RpcService.java

```diff
@@ -50,8 +50,12 @@ public boolean hasMethod(String method) {
     }
 
     private Object[] convertParams(Method method, JsonNode params) throws RpcException {
+        var optionalAnnotation = method.getAnnotation(RpcOptionalParameters.class);
+        var optionalParameters = optionalAnnotation == null ? 0 : optionalAnnotation.value();
         var parameters = method.getParameterTypes();
-        if (!params.isArray() || parameters.length != params.size()) {
+        if (!params.isArray() ||
+                params.size() > parameters.length ||
+                params.size() < parameters.length + optionalParameters) {
```

Comment:
`@RpcOptionalParameters(N)` allows the last N parameters to be omitted, they will be set to `null`.
Unfortunately, we can't just set default arguments on the methods because we're calling the methods via reflection here.
Good catch with check, that is the wrong way around yes.